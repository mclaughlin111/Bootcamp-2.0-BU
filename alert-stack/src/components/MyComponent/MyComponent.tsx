import { getFullName } from "../Player/getFullName";

type Name = {
  firstName: string;
  middleName?: string;
  lastName: string;
};

export type MyComponentProps = {
  name: Name;
  location: string;
  isEmployed: boolean;
};

export function MyComponent({ name, location, isEmployed }: MyComponentProps) {
  return (
    <>
      <div>{getFullName(name.firstName, name.lastName)}</div>
      <div>Location: {location}</div>
      <div>IsEmployed?: {isEmployed ? "Yes!" : "No"}</div>
    </>
  );
}
