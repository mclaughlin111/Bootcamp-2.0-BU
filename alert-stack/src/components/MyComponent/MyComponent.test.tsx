/* eslint-disable testing-library/no-debugging-utils */
import { render, screen } from "@testing-library/react";
import { MyComponent, MyComponentProps } from "./MyComponent";

const employees: MyComponentProps[] = [
  {
    name: { firstName: "Duncan", lastName: "Edwards" },
    isEmployed: true,
    location: "Bristol",
  },
  {
    name: { firstName: "Jane", lastName: "Smith" },
    isEmployed: false,
    location: "London",
  },
];

test.each<MyComponentProps>(employees)(
  "Render a basic my component",
  (employee) => {
    const { container } = render(<MyComponent {...employee} />);

    expect(container).toMatchSnapshot();
  }
);
