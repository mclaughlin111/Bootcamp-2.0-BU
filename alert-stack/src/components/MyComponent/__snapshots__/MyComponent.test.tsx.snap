// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a basic my component 1`] = `
<div>
  <div>
    Edwards, Duncan
  </div>
  <div>
    Location: 
    Bristol
  </div>
  <div>
    IsEmployed?: 
    Yes!
  </div>
</div>
`;

exports[`Render a basic my component 2`] = `
<div>
  <div>
    Smith, Jane
  </div>
  <div>
    Location: 
    London
  </div>
  <div>
    IsEmployed?: 
    No
  </div>
</div>
`;
